### PR TITLE
Vec 189 tls

### PR DESCRIPTION
**Pull request by jdogmcsteezy**
Add tls and user/pass authentication. This PR does not include managing the ACL.

### CODE REVIEW

File: cmd/utils.go [hunk 1]
```diff
@@ -2,10 +2,152 @@ package cmd
...
+	)
+	if err != nil {
+		logger.Error("failed to create AVS client", slog.Any("error", err))
+		return nil, err
+	}
+
+	return adminClient, nil
+}
+
+func newTLSConfig(rootCA [][]byte, cert []byte, key []byte, keyPass []byte, tlsProtoMin int, tlsProtoMax int) (*tls.Config, error) {
```
Comment (dwelch-spike) [inline cmd/utils.go:61]: Are this and the other TLS related functions below it used? I don't see a use of `newTLSConfig` .

**Review by dwelch-spike — COMMENTED** [review]
Nitpicks and wondering if `newTLSConfig` from utils.go is used? Otherwise, once you have had a look, LGTM. (ignoring tests and linter errors as discussed)

File: cmd/createIndex.go [hunk 2]
```diff
@@ -54,36 +50,34 @@ var createIndexFlags = &struct {
...
+	flagSet.StringToStringVar(&createIndexFlags.indexMeta, flags.IndexMeta, nil, commonFlags.DefaultWrapHelpString("The distance metric for the index."))                                                                                                                                                                                                                                        //nolint:lll // For readability
+	flagSet.DurationVar(&createIndexFlags.timeout, flags.Timeout, time.Second*5, commonFlags.DefaultWrapHelpString("The distance metric for the index."))                                                                                                                                                                                                                                        //nolint:lll // For readability
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply cmd/createIndex.go:61]: fixed later

File: cmd/createIndex.go [hunk 3]
```diff
@@ -54,36 +50,34 @@ var createIndexFlags = &struct {
...
+	flagSet.VarP(&createIndexFlags.distanceMetric, flags.DistanceMetric, "m", commonFlags.DefaultWrapHelpString(fmt.Sprintf("The distance metric for the index. Valid values: %s", strings.Join(flags.DistanceMetricEnum(), ", "))))                                                                                                                                                             //nolint:lll // For readability
+	flagSet.StringToStringVar(&createIndexFlags.indexMeta, flags.IndexMeta, nil, commonFlags.DefaultWrapHelpString("The distance metric for the index."))                                                                                                                                                                                                                                        //nolint:lll // For readability
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply cmd/createIndex.go:60]: fixed

File: cmd/dropIndex.go [hunk 4]
```diff
@@ -19,34 +18,29 @@ import (
...
-	flagSet.DurationVar(&dropIndexFlags.timeout, flagNameTimeout, time.Second*5, commonFlags.DefaultWrapHelpString("The distance metric for the index."))                                                                              //nolint:lll // For readability
+	flagSet.StringVarP(&dropIndexFlags.namespace, flags.Namespace, "n", "", commonFlags.DefaultWrapHelpString("The namespace for the index."))          //nolint:lll // For readability
+	flagSet.StringSliceVarP(&dropIndexFlags.sets, flags.Sets, "s", nil, commonFlags.DefaultWrapHelpString("The sets for the index."))                   //nolint:lll // For readability
+	flagSet.StringVarP(&dropIndexFlags.indexName, flags.IndexName, "i", "", commonFlags.DefaultWrapHelpString("The name of the index."))                //nolint:lll // For readability
+	flagSet.DurationVar(&dropIndexFlags.timeout, flags.Timeout, time.Second*5, commonFlags.DefaultWrapHelpString("The distance metric for the index.")) //nolint:lll // For readability
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply cmd/dropIndex.go:35]: fixed later. It moved somewhere else so can't fix now

File: cmd/utils.go [hunk 5]
```diff
@@ -2,10 +2,152 @@ package cmd
...
+	"golang.org/x/term"
+
+	avs "github.com/aerospike/avs-client-go"
 )
 
+func createClientFromFlags(clientFlags *flags.ClientFlags, connectTimeout time.Duration) (*avs.AdminClient, error) {
+	hosts, isLoadBalancer := parseBothHostSeedsFlag(clientFlags.Seeds, clientFlags.Host)
+
+	ctx, cancel := context.WithTimeout(context.Background(), connectTimeout)
+	defer cancel()
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply cmd/utils.go:24]: fixed later

File: cmd/utils.go (hunk 1 above)
Comment (jdogmcsteezy) [reply cmd/utils.go:61]: guess not. I must have copied that here before deciding to change the common lib

**Review by jdogmcsteezy — COMMENTED** [review]
replied

File: docker/auth/config/jwt/private_key.pem
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply docker/auth/config/jwt/private_key.pem:1]: yes

File: docker/auth/config/tls/localhost.keystore.jks
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply docker/auth/config/tls/localhost.keystore.jks:1]: yes

**Review by dwelch-spike — APPROVED** [review]
LGTM

**Pull request merged**